config/default/manager_metrics_patch.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
Why are we patching the deployment with the port vs not adding it into the deployment when its being created?

Reply:
TL;DR – Operator vs operand. It's a different Deployment.

The Operator's Deployment is created just once, when the Operator is installed. This is the Deployment whose missing port number this PR fixes. This Deployment won't ever be affected by reconciliation. It has nothing to do with Authorino CRs.

That said, we _also_ do not specify the port numbers in the Deployments (plural) that the Operator creates for the operand[^1][^2].

[^1]: https://github.com/Kuadrant/authorino-operator/blob/1acbdfc2040210a85a2e2cb156a22add964c1414/controllers/authorino_controller.go#L285

[^2]: https://github.com/Kuadrant/authorino-operator/blob/1acbdfc2040210a85a2e2cb156a22add964c1414/pkg/resources/k8s_deployment.go#L38-L44

The problems are unrelated though, even if they are perceived the same way by one who only wants to consume the metrics from all components.

We can fix the lack of port numbers in the operands' Deployments too (in another PR.)

Either way, let's keep in mind that changing the Deployments is not what will affect exposure of the port. From the docs[^3]:

[^3]: https://pkg.go.dev/k8s.io/api@v0.28.3/core/v1#Container

>  Not specifying a port here DOES NOT prevent that port from being exposed. Any port which is listening on the default "0.0.0.0" address inside a container will be accessible from the network.

Creating the Services is the main piece in our case. We currently create (and reconcile!) the metrics services for the operand all right. With this PR, we will be also creating (though no reconciliation needed) the metrics service for the Operator.

Reply:
Hahaha. Now I get what you mean!

> why not have the port in the deployment in the manager.yaml.

I suppose we could. Here I was the one following Limitador Operator's approach 🙃

Reply:
I feel like its an unnecessary step, i don't mind updating limitador to follow suit with authorino and update it as part of the service pr i, creating for limitador.

Reply:
Done.